src/core.ts

```diff
@@ -190,19 +191,11 @@ export const isolate = <M>(ctr: Container<M>, opts: any = {}): IsolatedContainer
  * will be treated as an updater.
  */
 export function seq<M>(...updaters: Updater<M>[]) {
-  return function (model: M, message?: GenericObject, relay?: GenericObject): UpdateResult<M> {
-    let newModel = model, commands = [], newCommands = [], updateResult = null;
-
-    for (const updater of updaters) {
-      updateResult = updater;
-
-      while (is(Function, updateResult)) {
-        updateResult = updateResult(newModel, message, relay);
-      }
-      [newModel, newCommands] = result(updateResult);
-      commands = flatten(commands.concat(newCommands));
-    }
-    return [newModel, commands] as UpdateResult<M>;
+  return function (model: M, msg?: GenericObject, relay?: GenericObject): UpdateResult<M> {
+    let reduce;
+    const merge = ([{}, cmds], [newModel, newCmds]) => [newModel, flatten(cmds.concat(newCmds))];
+    reduce = (prev, cur) => is(Function, cur) ? reduce(prev, cur(prev[0], msg, relay)) : merge(prev, mapResult(cur));
```

Comment:
why is reduce a `let` here?

Reply:
I think because the original definition was point-free.